source/Nevermore/Advanced/WriteTransaction.cs

```diff
@@ -217,5 +265,11 @@ public async Task CommitAsync(CancellationToken cancellationToken = default)
             await Transaction.CommitAsync(cancellationToken);
             await configuration.Hooks.AfterCommitAsync(this);
         }
+
+        static void AssertUpdateHasBeenSuccessful(int numberOfUpdatedRows, PreparedCommand command)
+        {
+            if (numberOfUpdatedRows == 0 && command.Mapping.RowVersionColumn != null) throw new StaleDataException("Update failed because submitted data was out of date. Refresh the data and try again.");
```

Comment:
Since we're checking the rows updated, do we want to throw a new exception if there's no `RowVersion` column but no row was updated?

Reply:
This would be a breaking change as Nevermore doesn't do that currently.